06x-new-kafka-roller.md

```diff
@@ -0,0 +1,350 @@
+ | LEADING_ALL_PREFERRED | Node is in running state and leading all preferred replicas. | None
+
+The definitions of broker states can be found via the following link: [Broker States](https://github.com/apache/kafka/blob/3.7/metadata/src/main/java/org/apache/kafka/metadata/BrokerState.java).
+
+The definitions of the possible restart reasons can be found via the following link: [Restart Reasons](https://github.com/strimzi/strimzi-kafka-operator/blob/0.40.0/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/RestartReason.java)
+
+#### NOT_RUNNING state
+
+If one of the following is true, then node's state is `NOT_RUNNING:
+- no pod exists for this node
+- unable to get the `Pod Status` for the pod 
+- the pod has `Pending` status with `Unschedulable` reason
+- the pod has container status `ContainerStateWaiting` with `CrashLoopBackOff` or `ImagePullBackOff` reason
+If none of the above is true but the node is not ready, then its state would be `NOT_READY`.
+
+#### Flow diagram describing the overall flow of the states
+![The new roller flow](./images/06x-new-roller-flow.png)
+
+
+### Configurability
+The following can be the configuration options for the new KafkaRoller:
+
+| Configuration          | Default value | Exposed to user | Description                                                                                                                                                                                                                                                   |
+|:-----------------------|:--------------|:----------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| maxRestartAttempts     | 3             | No              | The maximum number of times a node can be restarted before failing the reconciliation. This is checked against the node's `numRestartAttempts`.                                                                                                               |
+| maxReconfigAttempts    | 3             | No              | The maximum number of times a node can be reconfigured before restarting it. This is checked against the node's `numReconfigAttempts`.                                                                                                                        |
+| maxRetries             | 10            | No              | The maximum number of times a node can be retried after not meeting the safety conditions. This is checked against the node's `numRetries`.                                                                                                                      |
```

Comment:
I don't get this ... "retried" what? I mean, and sure it's me no being a native English, but what does exactly `The maximum number of times a node can be retried after not meeting the safety conditions` mean?

Reply:
I tried rephrasing this section, do you think it is clearer?

Reply:
what I see in the PR is now something like "The maximum number of times a node can be retried for  This is checked against the node's `numRetries`. " ... I think something is missing after "for".

Reply:
ah sorry, I updated the explanation on numRetries, but not this one. I will add that in.